app/components/primer/navigation/tab_component.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Primer
+  module Navigation
+    # This component is part of navigation components such as `Primer::TabNavComponent`
+    # and `Primer::UnderlineNavComponent` and should not be used as a standalone component.
+    class TabComponent < Primer::Component
+      include ViewComponent::SlotableV2
+
+      # Panel controlled by the Tab. It will only renderd if `with_panel` is `true`.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :panel, lambda { |**system_arguments|
+        system_arguments[:tag] ||= :div
+        system_arguments[:role] ||= :tabpanel
+        system_arguments[:hidden] = true unless @selected
+
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # Icon to be rendered in the Tab left.
+      #
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
+      renders_one :icon, lambda { |**system_arguments|
+        system_arguments[:classes] = class_names(
+          @icon_classes,
+          system_arguments[:classes]
+        )
+        Primer::OcticonComponent.new(**system_arguments)
+      }
+
+      # The Tab's title.
+      #
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::TextComponent) %>.
+      renders_one :title, Primer::TextComponent
```

Comment:
What do you think about naming this something else? `title` is a valid argument to `content_tag` and thus might be nice to have as an option here.

This brings up a more general concern: I don't think we have a solid set of reserved words established for this library. Perhaps we could look into that soon?

Reply:
that's a good idea!

Reply:
`text`?